src/m365/aad/commands/administrativeunit/administrativeunit-member-add.ts

```diff
@@ -0,0 +1,166 @@
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        userId: typeof args.options.userId !== 'undefined',
+        userName: typeof args.options.userName !== 'undefined',
+        groupId: typeof args.options.groupId !== 'undefined',
+        groupName: typeof args.options.groupName !== 'undefined',
+        deviceId: typeof args.options.deviceId !== 'undefined',
+        deviceName: typeof args.options.deviceName !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-i, --administrativeUnitId [administrativeUnitId]'
+      },
+      {
+        option: '-n, --administrativeUnitName [administrativeUnitName]'
+      },
+      {
+        option: "--userId [userId]"
+      },
+      {
+        option: "--userName [userName]"
+      },
+      {
+        option: "--groupId [groupId]"
+      },
+      {
+        option: "--groupName [groupName]"
+      },
+      {
+        option: "--deviceId [deviceId]"
+      },
+      {
+        option: "--deviceName [deviceName]"
+      }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.administrativeUnitId && !validation.isValidGuid(args.options.administrativeUnitId as string)) {
+          return `${args.options.administrativeUnitId} is not a valid GUID`;
+        }
+
+        if (args.options.userId && !validation.isValidGuid(args.options.userId as string)) {
+          return `${args.options.userId} is not a valid GUID`;
+        }
+
+        if (args.options.groupId && !validation.isValidGuid(args.options.groupId as string)) {
+          return `${args.options.groupId} is not a valid GUID`;
+        }
+
+        if (args.options.deviceId && !validation.isValidGuid(args.options.deviceId as string)) {
+          return `${args.options.deviceId} is not a valid GUID`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push({ options: ['administrativeUnitId', 'administrativeUnitName'] });
+    this.optionSets.push({ options: ['userId', 'userName', 'groupId', 'groupName', 'deviceId', 'deviceName'] });
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let administrativeUnitId = args.options.administrativeUnitId;
+    let memberType;
+    let memberId;
+
+    try {
+      if (args.options.administrativeUnitName) {
+        administrativeUnitId = (await aadAdministrativeUnit.getAdministrativeUnitByDisplayName(args.options.administrativeUnitName)).id!;
+      }
+
+      if (args.options.userId || args.options.userName) {
+        memberType = 'users';
+        memberId = args.options.userId;
+        if (args.options.userName) {
+          memberId = await aadUser.getUserIdByUpn(args.options.userName);
```

Comment:
Let's add a verbose logging line about retrieving the user id. Let's add verbose logging lines on all these kinds of places